app/uk/gov/hmrc/bindingtariffclassification/repository/CaseRepository.scala

```diff
@@ -65,17 +65,14 @@ class CaseMongoRepository @Inject()(mongoDbProvider: MongoDbProvider)
   }
 
   override def update(c: Case): Future[Option[Case]] = {
-    atomicUpdate(SearchCase(reference = Some(c.reference)).buildJson, c)
+    atomicUpdate(jsonMapper.fromReference(c.reference), c)
   }
 
   override def getByReference(reference: String): Future[Option[Case]] = {
-    getOne(SearchCase(reference = Some(reference)).buildJson)
+    getOne(jsonMapper.fromReference(reference))
   }
 
-  override def get(searchBy: Option[SearchCase] = None, sortedBy: Option[SortCase] = None): Future[Seq[Case]] = {
-    val searchFrom = searchBy.map(_.buildJson).getOrElse(Json.obj())
-    val sortFrom = sortedBy.map(_.buildJson).getOrElse(Json.obj())
-
-    getMany(searchFrom, sortFrom)
+  override def get(searchBy: CaseParamsFilter, sortedBy: Option[String]): Future[Seq[Case]] = {
+    getMany(jsonMapper.from(searchBy), Json.obj())
```

Comment:
This is confusing.
You define the `sortedBy` param, but you do not use it.

I would expect something like this:
```
val sorting = sortedBy match {
  case Some(field) => ??? // TODO
  case None => Json.obj()
}

getMany(jsonMapper.from(searchBy), sorting)
```